sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerContext.java

```diff
@@ -191,6 +204,17 @@ public DataContext createDataContext(final JavaTypeFactory typeFactory)
           DataContext.Variable.TIME_ZONE.camelName, localNow.getZone().toTimeZone().clone(),
           DATA_CTX_AUTHENTICATION_RESULT, authenticationResult
       );
+      private final Map<String, Object> context;
+
+      DruidDataContext()
+      {
+        ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
+        builder.putAll(base_context);
+        for (int i = 0; i < parameters.size(); i++) {
```

Comment:
Depending on the implementation and size of the list, it may be more efficient to iterate directly over the list values, e.g.:
```java
int i = 0;
for (TypedValue parameter : parameters) {
    builder.put("?" + i, parameter.value);
    i++;
}
```

Reply:
I doubt this matters much at the size this will likely be, but can change

Reply:
changed